modules/modules.cmake

```diff
@@ -0,0 +1,19 @@
+# SPDX-License-Identifier: Apache-2.0
+
+file(GLOB cmake_modules "${CMAKE_CURRENT_LIST_DIR}/*/CMakeLists.txt")
```

Comment:
Should this be looking into all MODULE_ROOTs? I.e. not just `${CMAKE_CURRENT_LIST_DIR}`? Or does each module_root need to contain a copy of this modules.cmake file?

Reply:
They need to provide their own `modules.cmake`.

How they decide to structure module glue code is free for them to decide.
They can loop folders like Zephyr does, or use a specific list, or even point to a third location.

Reply:
This is described in the added docs.